e2e/vmss.go

```diff
@@ -54,7 +54,14 @@ func bootstrapVMSS(ctx context.Context, t *testing.T, r *mrand.Rand, vmssName st
 }
 
 func createVMSSWithPayload(ctx context.Context, customData, cseCmd, vmssName string, publicKeyBytes []byte, opts *scenarioRunOpts) (*armcompute.VirtualMachineScaleSet, error) {
-	model := getBaseVMSSModel(vmssName, opts.suiteConfig.location, opts.suiteConfig.subscription, *opts.clusterConfig.cluster.Properties.NodeResourceGroup, opts.clusterConfig.subnetId, string(publicKeyBytes), customData, cseCmd)
+	model := getBaseVMSSModel(vmssName, string(publicKeyBytes), customData, cseCmd, opts)
+
+	if opts.suiteConfig.buildID != "" {
```

Comment:
can this be nil?

Reply:
it should be able to nil since we construct it before running all scenarios, though maybe in a separate PR we can add proto-like getters to ensure we safely access these fields

Reply:
I love proto like getters.